### DISCUSSION TITLE
Event that is triggered when creating xforms?

### DISCUSSION THREAD

**Discussion by gracekumagai** [Autodesk/maya-usd#1952] [Q&A] [answered]
Hello! I was wondering if there is an event that is triggered when xforms are created that we can use to create xformOps for translate, rotate and scale on the new prim so the attributes will appear (similar to native maya) in the Channel Box without needing to manipulate them first? 

We are hoping to create the xformOps when:
- An artist creates an empty xform
- An artist or script creates a group

Thanks!

**Comment by seando-adsk** [comment, marked as answer]
Hi Grace, are you asking if there is a Maya event triggered because the answer would be no. In the MayaUsd lib we have a [StagesSubject](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/ufe/StagesSubject.cpp) that gets all the Usd notices and translates them into Ufe which it then sends out. From Ufe you will be a generic ObjectAdd which you could then query to see if it was an xform.

Sean

**Reply by gracekumagai** [reply]
Thank you!
